### PR TITLE
Update change password in `<UserProfile />`

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [x] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

- We take advantage of the new updatePassword method from UserResource
- Introduced a new text field for the current password of a user (if that exists)
- Introduced a new checkbox that invalidates all other sessions of the user when selected
- New localization keys to support the introduced labels

### Change password flow (session sign out)
https://user-images.githubusercontent.com/19269911/229761051-15a928d8-3584-4102-af18-c05317158026.mp4
### Change password flow
https://user-images.githubusercontent.com/19269911/229761169-bebdfba3-7a4d-48b7-84a0-fc98194210d8.mp4
### Set password flow (session sign out)
https://user-images.githubusercontent.com/19269911/229761237-8ac8577b-f3fe-4d0a-a9b2-bc0a30afdfb1.mp4

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx [hunk 1]
```diff
@@ -34,14 +57,47 @@ export const PasswordPage = withCardStateProvider(() => {
   };
 
   const updatePassword = async () => {
+    const opts: Parameters<typeof user.updatePassword>[0] = {
+      newPassword: passwordField.value,
+      signOutOfOtherSessions: sessionsField.checked,
+      currentPassword: undefined,
```
Comment (raptisj) [inline packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx:63]: @panteliselef 🙃  couldn’t this be:
```
currentPassword: user.passwordEnabled ? currentPasswordField.value : undefined

File: packages/clerk-js/src/ui/components/UserProfile/__tests__/PasswordPage.test.tsx [hunk 2]
```diff
@@ -11,6 +10,10 @@ const initConfig = createFixtures.config(f => {
   f.withUser({});
 });
 
+const changePasswordConfig = createFixtures.config(f => {
+  f.withUser({ password_enabled: true });
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/UserProfile/__tests__/PasswordPage.test.tsx:14]: I think there's also `f.withPassword()`

Comment (panteliselef) [reply packages/clerk-js/src/ui/components/UserProfile/__tests__/PasswordPage.test.tsx:14]: `f.withPassword({ required: true, enabled: true })` does not seem to work tho, the tests will fails when they shouldn't

**Comment by desiprisg** [comment]
Why is the account tab highlighted while the breadcrumb says "Security". Is that another bug or did something break here?

File: packages/localizations/src/en-US.ts [hunk 3]
```diff
@@ -24,8 +24,10 @@ export const enUS: LocalizationResource = {
   formFieldLabel__phoneNumber_username: 'phone number or username',
   formFieldLabel__emailAddress_phoneNumber_username: 'Email address, phone number or username',
   formFieldLabel__password: 'Password',
+  formFieldLabel__currentPassword: 'Current Password',
```
Comment (anagstef) [inline packages/localizations/src/en-US.ts:27]:
```suggestion
  formFieldLabel__currentPassword: 'Current password',
```

File: packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx [hunk 4]
```diff
@@ -21,7 +53,14 @@ export const PasswordPage = withCardStateProvider(() => {
...
   });
 
-  const canSubmit = passwordField.value && passwordField.value === confirmField.value && passwordField.value.length > 7;
+  const sessionsField = useFormControl('signOutOfOtherSessions', '', {
+    type: 'checkbox',
+    label: localizationKeys('formFieldLabel__signOutOfOtherSessions'),
+  });
+
+  const isPasswordMatch =
+    passwordField.value && passwordField.value === confirmField.value && passwordField.value.length > 7;
```
Comment (anagstef) [inline packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx:62]: ❓ What is this `7` here? Password needs more than 7 characters?

**Review by anagstef — COMMENTED** [review]
Nice work!

File: packages/clerk-js/src/ui/utils/useFormControl.ts [hunk 5]
```diff
@@ -26,26 +27,34 @@ type FieldStateProps<Id> = {
...
 
-export const useFormControl = <Id extends string>(
+export const useFormControl = <Id extends string, TOptions extends Options = any>(
   id: Id,
   initialState: string,
-  opts?: Options,
+  opts?: TOptions,
 ): FormControlState<Id> => {
-  opts = opts || { type: 'text', label: '', isRequired: false, placeholder: '', options: [] };
+  opts = opts || ({ type: 'text', label: '', isRequired: false, placeholder: '', options: [] } as unknown as TOptions);
```
[the remark this answers is not in the document]
Comment (panteliselef) [reply packages/clerk-js/src/ui/utils/useFormControl.ts:39]: I believe we can, I'll have another look at this one.

**Comment by panteliselef** [comment]
> Why is the account tab highlighted while the breadcrumb says "Security"? Is that another bug or did something break here?

@desiprisg  good catch, but this seems to be a pre-existring issue. Login and check it here   https://lazy-clerk.vercel.app/user

File: packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx (hunk 4 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx:62]: @anagstef Yes, clerk requires at least 8 characters for a user's password.

Comment (anagstef) [reply packages/clerk-js/src/ui/components/UserProfile/PasswordPage.tsx:62]: @panteliselef Can we extract this `7` somewhere? It's getting lost here. We can introduce an `isValidPassword` method and a static const like `PASSWORD_MIN_LENGTH`, or something similar.

File: packages/localizations/src/en-US.ts (hunk 3 above)
Comment (anagstef) [reply packages/localizations/src/en-US.ts:27]: @panteliselef I still see the `Password` capitalized. Do we want to keep this, or you missed it?

**Review by desiprisg — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**